crates/compiler/mono/src/ir.rs

```diff
@@ -9268,9 +9268,13 @@ fn assign_num_literal_expr<'a>(
     num_value: IntOrFloatValue,
     hole: &'a Stmt<'a>,
 ) -> Stmt<'a> {
-    let layout = layout_cache
-        .from_var(env.arena, variable, env.subs)
-        .unwrap();
+    let layout = match layout_cache.from_var(env.arena, variable, env.subs) {
+        Ok(layout) => layout,
+        Err(_) => match num_value {
+            IntOrFloatValue::Float(_) => Layout::default_float(),
+            IntOrFloatValue::Int(_) => Layout::default_integer(),
+        },
```

Comment:
I don't think this is what we want to do, because then the program will run, but with the wrong number value. Instead, to fix this issue, this function should be changed to return `Result<Stmt<'a>, RuntimeError>`. Then the type error will bubble up and become a runtime error at the Roc function level.

Reply:
Thank you. I'll fix it.